src/tqecd/match_utils/__init__.py

```diff
@@ -0,0 +1,27 @@
+"""Provides a few utility functions to match flows.
```

Comment:
I'm not sure if the phrase "match flows" is appropriate; in ML, there's also a concept called flow matching, which could confuse some people.